libraries/ESP8266WebServer/src/Parsing.cpp

```diff
@@ -111,6 +201,19 @@ bool ESP8266WebServer::_parseRequest(WiFiClient& client) {
   }
   _currentMethod = method;
 
+  //allows http redirection with URL of the form:
+  //http://SERVERIP/fwd/REDIRECTIP --> very useful in meshnetwork architecture
+  //http://192.168.0.7/fwd/192.168.4.2 --> for instance, this allows to connect the chip with IP 192.168.4.2 which is connected to the softAP of node 192.168.0.7  
+  // (POST requests are also supported)
+  if(url.indexOf("/fwd/") == 0){
```

Comment:
This certainly has to be an optional feature. You can implement it as a library which provides a class derived from a RequestHandler, instead of plugging directly into the server class.